components/brave_wallet/resources/brave_wallet_provider.js

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function() {
+    if (!window.ethereum) {
+      return
+    }
+
+    // TODO: fill window.ethereum object with functions
```

Comment:
?

Reply:
we're defining this in c++, why do we need to execute any JS at all?

Reply:
removed scripts execution